back/src/dals/user/repository/user.repository.ts

```diff
@@ -0,0 +1,92 @@
+import {
+  UserSessionContext,
+  UserSession,
+  ConnectSessionInfo,
+  VotesFromRooms,
+} from 'dals/user';
+//This is for store the session Info in mongo database.
+
+export const isRoomAvailable = async (room: string): Promise<Boolean> => {
+  return !(await UserSessionContext.exists({ room }));
+};
+
+export const addNewUser = async (
+  connectionId: string,
+  connectSession: ConnectSessionInfo
+): Promise<UserSession> => {
+  const data = {
+    connectionId,
+    ...connectSession,
+    hasVoted: false,
+    vote: '',
+  };
+  await UserSessionContext.create(data).catch(console.log);
+  return UserSessionContext.findOne({ connectionId }).lean();
+};
+
+export const isMasterUser = async (
+  connectionId: string
+): Promise<UserSession> =>
+  await UserSessionContext.findOne({ connectionId, isMaster: true }).lean();
+
+export const isNicknameInUse = async (
+  nickname: string,
+  room: string
+): Promise<Boolean> => await UserSessionContext.exists({ nickname, room });
+
+export const getRoomFromConnectionId = async (
+  connectionId: string
+): Promise<string> => {
+  const session = await UserSessionContext.findOne({
+    connectionId: connectionId,
+  }).lean();
+  return session ? session.room : '';
+};
+
+export const getNicknameFromConnectionId = async (
+  connectionId: string
+): Promise<string> => {
+  const session = await UserSessionContext.findOne({
+    connectionId: connectionId,
+  }).lean();
+
+  return session ? session.nickname : '';
+};
+
+export const resetVotes = async (room: string): Promise<void> => {
+  return await UserSessionContext.findOneAndUpdate(
```

Comment:
You are returning the user, but the return type is `Promise<void>`. Do we need return something?

Reply:
nope we don't need